### PR TITLE
OpenStack: Rename lbFloatingIP to apiFloatingIP

### PR DESCRIPTION
**Pull request by Fedosin**
/label platform/openstack

### CODE REVIEW

**Review by mandre — COMMENTED** [review]
/lgtm

File: data/data/install.openshift.io_installconfigs.yaml [hunk 1]
```diff
@@ -998,6 +998,9 @@ spec:
               openstack:
                 description: OpenStack is the configuration used when installing on OpenStack.
                 properties:
+                  apiFloatingIP:
+                    description: APIFloatingIP is the IP address of an available floating IP in your OpenStack cluster to associate with the OpenShift API.
```
Comment (wking) [inline data/data/install.openshift.io_installconfigs.yaml:1002]: nit: `with the OpenShift API.` -> `with the OpenShift API load balancer.`?

File: docs/user/openstack/customization.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ platform:
     externalDNS:
       - "8.8.8.8"
       - "192.168.1.12"
-    lbFloatingIP: 128.0.0.1
+    apiFloatingIP: 128.0.0.1
```
Comment (wking) [inline docs/user/openstack/customization.md:74]: nit: re-alphabetize, now that you start with `a`, here and below?

File: pkg/types/conversion/installconfig.go [hunk 3]
```diff
@@ -26,19 +27,21 @@ func ConvertInstallConfig(config *types.InstallConfig) error {
...
 	}
 
 	config.APIVersion = types.InstallConfigVersion
 	return nil
 }
 
-// ConvertNetworking upconverts deprecated fields in networking
-func ConvertNetworking(config *types.InstallConfig) {
+// convertNetworking upconverts deprecated fields in networking
+func convertNetworking(config *types.InstallConfig) {
```
Comment (wking) [inline pkg/types/conversion/installconfig.go:44]: I dunno why this was public before, but making it private seems like an orthogonal change that should go in its own PR.

File: pkg/types/conversion/installconfig.go [hunk 4]
```diff
@@ -79,11 +82,19 @@ func ConvertNetworking(config *types.InstallConfig) {
...
+func convertBaremetal(config *types.InstallConfig) {
 	if config.Platform.BareMetal.DeprecatedProvisioningDHCPExternal == true && config.Platform.BareMetal.ProvisioningNetwork == "" {
 		config.Platform.BareMetal.ProvisioningNetwork = baremetal.UnmanagedProvisioningNetwork
 	}
 }
+
+// convertOpenStack upconverts deprecated fields in the OpenStack platform.
+func convertOpenStack(config *types.InstallConfig) {
+	// LbFloatingIP has been renamed to APIFloatingIP
+	if config.Platform.OpenStack.DeprecatedLbFloatingIP != "" && config.Platform.OpenStack.APIFloatingIP == "" {
```
Comment (wking) [inline pkg/types/conversion/installconfig.go:97]: Seems like it should be a fatal error to have both set.  Also, probably worth warn-level logging when you do shuffle the value over, so folks using the old value are more likely to hear about the deprecation and update their generators.

File: pkg/types/conversion/installconfig.go [hunk 5]
```diff
@@ -79,11 +82,19 @@ func ConvertNetworking(config *types.InstallConfig) {
...
+		config.Platform.OpenStack.APIFloatingIP = config.Platform.OpenStack.DeprecatedLbFloatingIP
```
Comment (wking) [inline pkg/types/conversion/installconfig.go:98]: I'd expect to clear the old property when you shuffle the value over.

File: pkg/types/conversion/installconfig.go (hunk 4 above)
Comment (Fedosin) [reply pkg/types/conversion/installconfig.go:97]: this will go in a separate PR for all deprecated values. we have a dedicated jira card for this.

File: data/data/install.openshift.io_installconfigs.yaml (hunk 1 above)
Comment (Fedosin) [reply data/data/install.openshift.io_installconfigs.yaml:1002]: done

File: pkg/types/conversion/installconfig.go (hunk 5 above)
Comment (Fedosin) [reply pkg/types/conversion/installconfig.go:98]: I think we should keep it to show a warning message later, during install config validation.

File: pkg/types/conversion/installconfig.go (hunk 3 above)
Comment (Fedosin) [reply pkg/types/conversion/installconfig.go:44]: okay, I'll propose another PR to fix this.

File: pkg/asset/installconfig/openstack/validation/platform.go [hunk 6]
```diff
@@ -106,13 +106,13 @@ func validatePlatformFlavor(p *openstack.Platform, ci *CloudInfo, fldPath *field
 }
 
 func validateFloatingIPs(p *openstack.Platform, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/openstack/validation/platform.go:108]: can we add a validation that makes sure people don't set both with different values.

File: pkg/types/conversion/installconfig.go (hunk 5 above)
Comment (abhinavdahiya) [reply pkg/types/conversion/installconfig.go:98]: > I'd expect to clear the old property when you shuffle the value over.

Don't reset user set values, if a user set the deprecated value, we can use that to set the new values but do not modify the value set by the user.

**Comment by abhinavdahiya** [comment]
> https://github.com/openshift/installer/pull/4244/files#r501164253

LGTM, only one change requested.

File: pkg/asset/installconfig/openstack/validation/platform.go (hunk 6 above)
Comment (Fedosin) [reply pkg/asset/installconfig/openstack/validation/platform.go:108]: I added a check that returns an error if both values are set regardless of whether they are equal or not. I.e. "you cannot specify lbFloatingIP and apiFloatingIP in the install config together".

**Review by pierreprinetti — APPROVED** [review]

**Comment by pierreprinetti** [comment]
/hold
Do you want to squash your commits?

**Comment by Fedosin** [comment]
/hold cancel

nope :)
